### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by marcelorl** [puphpet/puphpet#1773]
When I try to access any of the virtual hosts I've created
 - test1.project.lan
 - test2.project.lan
 - test3.project.lan
 - test4.project.lan

I get the fucking Internal Server Error.

I'm stuck in this issue. Unfortunately I'll need to install a local environment :'(. I have already another 2 applications using homestead, so at least vagrant is working in my machine with no problems.

My config.yaml:

```vagrantfile:
    target: local
    vm:
        box: puphpet/ubuntu1204-x64
        box_url: puphpet/ubuntu1204-x64
        hostname: local.puphpet2
        memory: '512'
        cpus: '1'
        chosen_provider: virtualbox
        network:
            private_network: 192.168.56.101
            forwarded_port:
                vflnp_stuhtuli2lt6:
                    host: '6020'
                    guest: '22'
        post_up_message: ''
        provider:
            virtualbox:
                modifyvm:
                    natdnshostresolver1: 'on'
                showgui: '0'
            vmware:
                numvcpus: 1
            parallels:
                cpus: 1
        provision:
            puppet:
                manifests_path: puphpet/puppet
                manifest_file: site.pp
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
                    - '--parser future'
        synced_folder:
            vflsf_75ooqiubhfp0:
                source: ./
                target: /var/www
                sync_type: default
                smb:
                    smb_host: ''
                    smb_username: ''
                    smb_password: ''
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                        - .git/
                    auto: 'true'
                owner: www-data
                group: www-data
        usable_port_range:
            start: 10200
            stop: 10500
    ssh:
        host: null
        port: null
        private_key_path: null
        username: vagrant
        guest_port: null
        keep_alive: true
        forward_agent: false
        forward_x11: false
        shell: 'bash -l'
    vagrant:
        host: detect
server:
    install: '1'
    packages:
        - vim
users_groups:
    install: '1'
    groups: {  }
    users: {  }
locale:
    install: '1'
    settings:
        default_locale: ''
        locales:
            - 'en_GB.UTF-8 UTF-8'
            - 'en_US.UTF-8 UTF-8'
firewall:
    install: '1'
    rules: {  }
cron:
    install: '1'
    jobs: {  }
nginx:
    install: '0'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
    upstreams: {  }
    vhosts:
        nxv_de09eols48oy:
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
            ssl_protocols: ''
            ssl_ciphers: ''
            rewrite_to_https: '1'
            spdy: '1'
            locations:
                nxvl_2ilr7o6e8ft6:
                    location: /
                    autoindex: 'off'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                nxvl_7o8t7v9mj6po:
                    location: '~ \.php$'
                    autoindex: 'off'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $request_filename'
                        - 'APP_ENV dev'
    proxies: {  }
apache:
    install: '1'
    settings:
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - proxy_fcgi
        - rewrite
    vhosts:
        av_ikxap0i30bsl:
            servername: test1.project.lan
            serveraliases:
                - test1.project.lan
            docroot: /var/www/project/sites/test1/webroot
            port: '80'
            setenv:
                - 'APP_ENV dev'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_pguh2w7ailu1:
                    path: /var/www/project/sites/test1/webroot
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_5vx8rh2nvbkr:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
        av_110grqt2xpzq:
            servername: test2.project.lan
            serveraliases:
                - test2.project.lan
            docroot: /var/www/project/sites/test2/webroot
            port: '80'
            setenv:
                - 'APP_ENV dev'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_0tg5lz0wo273:
                    path: /var/www/project/sites/test2/webroot
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_5vx8rh2nvbkr:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
        av_j7e8we8g6i3x:
            servername: test3.project.lan
            serveraliases:
                - test3.project.lan
            docroot: /var/www/project/sites/test3/webroot
            port: '80'
            setenv:
                - 'APP_ENV dev'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_m0d2r43nbxfe:
                    path: /var/www/project/sites/test3/webroot
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_5vx8rh2nvbkr:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
        av_zx8bh0letljm:
            servername: test4.project.lan
            serveraliases:
                - test4.project.lan
            docroot: /var/www/project/sites/test4/webroot
            port: '80'
            setenv:
                - 'APP_ENV dev'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_unezw0iuk5da:
                    path: /var/www/project/sites/test4/webroot
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_5vx8rh2nvbkr:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
php:
    install: '1'
    settings:
        version: '56'
    modules:
        php:
            - cli
            - intl
            - mcrypt
        pear: {  }
        pecl:
            - pecl_http
    ini:
        display_errors: 'On'
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
        date.timezone: UTC
    fpm_ini:
        error_log: /var/log/php-fpm.log
    fpm_pools:
        phpfp_5ynt77e6leeq:
            ini:
                prefix: www
                listen: '127.0.0.1:9000'
                security.limit_extensions: .php
                user: www-user
                group: www-data
    composer: '1'
    composer_home: ''
xdebug:
    install: '0'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
blackfire:
    install: '0'
    settings:
        server_id: ''
        server_token: ''
        agent:
            http_proxy: ''
            https_proxy: ''
            log_file: stderr
            log_level: '1'
        php:
            agent_timeout: '0.25'
            log_file: ''
            log_level: '1'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.19.0
drush:
    install: '0'
    version: 6.3.0
ruby:
    install: '1'
    versions: {  }
python:
    install: '1'
    packages: {  }
    versions: {  }
nodejs:
    install: '0'
    npm_packages: {  }
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings: {  }
    server_ini:
        hhvm.server.host: 127.0.0.1
        hhvm.server.port: '9000'
        hhvm.log.use_log_file: '1'
        hhvm.log.file: /var/log/hhvm/error.log
    php_ini:
        display_errors: 'On'
        error_reporting: '-1'
        date.timezone: UTC
mysql:
    install: '1'
    settings:
        version: '5.6'
        root_password: '123'
        override_options: {  }
    adminer: '1'
    users:
        mysqlnu_ig8hwx0r1ulk:
            name: dbuser
            password: '123'
    databases:
        mysqlnd_rrmzkh7r1upl:
            name: dbname
            sql: ''
    grants:
        mysqlng_92ejcljlhgh2:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
mariadb:
    install: '0'
    settings:
        version: '10.0'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mariadbnu_04h46zb8zhfh:
            name: dbuser
            password: '123'
    databases:
        mariadbnd_7nd5okgegh64:
            name: dbname
            sql: ''
    grants:
        mariadbng_g4kuhbp2e6n1:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.3'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
mongodb:
    install: '0'
    settings:
        auth: 1
        bind_ip: 127.0.0.1
        port: '27017'
    databases: {  }
redis:
    install: '0'
    settings:
        conf_port: '6379'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mailcatcher:
    install: '1'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '1080'
        mailcatcher_path: /usr/local/rvm/wrappers/default
        from_email_method: inline
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '13000'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
rabbitmq:
    install: '0'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 1.4.1
        java_install: true
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'
```

Generally an Internal Server Error means the webroot folder has no .htaccess file, but in my case, all of them have a .htaccess:

```
Options -MultiViews +FollowSymLinks

RewriteEngine On
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^(.+)$ /index.php?dispatch=$1&%{QUERY_STRING} [L]

php_flag register_globals Off
php_flag magic_quotes_gpc Off
php_flag track_vars On

<FilesMatch ".*\.(ico|pdf|flv|jpe?g|png|gif|js|css|swf|demo)$">
    Header set Cache-Control "public"
    ExpiresActive On
    ExpiresDefault "access plus 1 years"
    FileETag None
</FilesMatch>

<FilesMatch ".*\.(js|css)$">
    SetOutputFilter DEFLATE
#    FileETag MTime Size
</FilesMatch>
```

I have no idea why it is not working. I would be glad with any help of you guys. Thanks.

**Comment by marcelorl** [comment]
I've checked my log errors under /var/log/apache2, and I found this:


[Tue Jul 14 14:06:36.635967 2015] [core:alert] [pid 1647:tid 140406582871808] [client 192.168.56.1:54136] /var/www/project/sites/test4/webroot/.htaccess: Invalid command 'php_flag', perhaps misspelled or defined by a module not included in the server configuration, referer: http://test4.project.lan/


What do you guys think?

**Comment by jtreminio** [comment]
`php_flag` doesn't work under fcgi, which is what puphpet uses. You should consider setting those values into the php.ini itself, using the gui section.

**Comment by marcelorl** [comment]
uuuuu, maybe the fastest fix from here.

I finally fixed the problem.
Actually it was an issue very specific to my project, all related to the .htaccess above. The fix was:

Remove the following block from .htaccess file:
```
php_flag register_globals Off
php_flag magic_quotes_gpc Off
php_flag track_vars On
```

And paste it in a php.ini.

Then, I got another error related to the header. 
`Header set Cache-Control "public"`

I just ran:
```
a2enmod headers
service apache2 restart
```

Then, I got another fucking error, "maybe" related to memcache (my project uses memcache).
And I followed this tutorial:
https://www.digitalocean.com/community/tutorials/how-to-install-and-use-memcache-on-ubuntu-12-04
And had to run these commands:
```
LoadModule expires_module libexec/mod_expires.so
service apache2 restart
```




And finally it all over.



But I would love to know if there is a way to set these kind of configs within config.yaml... Would it be possible?

Thanks.
